libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -112,12 +112,20 @@ AP_GPS_SBF::read(void)
                                 }
                                 break;
                             case Config_State::SSO:
-                                if (asprintf(&config_string, "sso,Stream%d,COM%d,PVTGeodetic+DOP+ReceiverStatus+VelCovGeodetic+BaseVectorGeod,msec100\n",
-                                             (int)GPS_SBF_STREAM_NUMBER,
-                                             (int)gps._com_port[state.instance]) == -1) {
-                                    config_string = nullptr;
-                                }
-                                break;
+                                if (get_type() == AP_GPS::GPS_TYPE_SBF) {
+                                    if (asprintf(&config_string, "sso,Stream%d,COM%d,PVTGeodetic+DOP+ReceiverStatus+VelCovGeodetic+BaseVectorGeod,msec100\n",
```

Comment:
The alternate form of this that would use less flash is to pass the extra arguments as a %s at the end of the string like this: `"sso,Stream%d,COM%d,PVTGeodetic+DOP+ReceiverStatus+VelCovGeodetic+BaseVectorGeod%s,msec100\n"` then pass a string that is either empty "" or contains "AttEuler+AttCovEuler" depending on which type it is (which can be done as a separate switch before the asprintf call.  This is both quite readable, reduces flash cost, and most importantly it makes it easier to add (or remove) more required terms from all the driver cases, without having to duplicate it into every string manually.